.changeset/long-queens-draw.md

```diff
@@ -0,0 +1,62 @@
+---
+"thirdweb": minor
+---
+
+Added new deployment utility functions to help manage infrastructure contracts and initialization:
+
+- `getDeployedCloneFactoryContract`: Get the deployed clone factory contract instance
+- `getDeployedInfraContract`: Get or deploy infrastructure contracts like WETH, Forwarder etc
+- `getAllDefaultConstructorParamsForImplementation`: Get default constructor params for contract implementations
+- `getInitializeTransaction`: Prepare initialization transaction for contract deployment
+- `getOrDeployInfraForPublishedContract`: Get or deploy required infrastructure for published contracts
+
+```typescript
+import { 
+  getDeployedCloneFactoryContract,
+  getDeployedInfraContract,
+  getAllDefaultConstructorParamsForImplementation,
+  getInitializeTransaction,
+  getOrDeployInfraForPublishedContract
+} from "thirdweb";
+
+// Get deployed clone factory
+const factory = await getDeployedCloneFactoryContract({
+  client,
+  chain
+});
+
+// Get or deploy WETH contract
+const weth = await getDeployedInfraContract({
+  client,
+  chain,
+  contractId: "WETH9"
+});
+
+// Get default constructor params
+const params = await getAllDefaultConstructorParamsForImplementation({
```

Comment:
can we keep this one un-exported? i actually want to get rid of it in favor of what yash is working on